### PR TITLE
Actually call the solver in the parallel AMGX test

### PR DESCRIPTION
**Pull request by masterleinad**
I needed to adjust the tolerance of the outer solver to make the test pass for me locally.

### CODE REVIEW

File: tests/test_amgx_direct_solver.cu [hunk 1]
```diff
@@ -176,7 +176,8 @@ BOOST_AUTO_TEST_CASE(amgx_2_procs)
         cusparseSetMatIndexBase(matrix_dev->descr, CUSPARSE_INDEX_BASE_ZERO);
     mfmg::ASSERT_CUSPARSE(cusparse_error_code);
     auto rhs_dev = mfmg::copy_from_host(rhs);
-    auto solution_dev = mfmg::copy_from_host(sol_ref);
+    auto solution_dev = mfmg::copy_from_host(rhs);
+    solution_dev = 0.;
```
Comment (dalg24) [inline tests/test_amgx_direct_solver.cu]: Was it wrong to copy `sol_ref`?  They seem to be both built with the same index range.  Was the issue initialization to zero?  I thought deal.II vector constructors where usually doing the initialization to zero, so copying from `rhs` would already have the zeroes.

File: tests/test_amgx_direct_solver.cu [hunk 2]
```diff
@@ -190,6 +191,7 @@ BOOST_AUTO_TEST_CASE(amgx_2_procs)
     mfmg::CudaSolver<dealii::LinearAlgebra::distributed::Vector<
         double, dealii::MemorySpace::CUDA>>
         direct_solver_dev(cuda_handle, cuda_op, params);
+    direct_solver_dev.apply(rhs_dev, solution_dev);
```
Comment (dalg24) [inline tests/test_amgx_direct_solver.cu]: Weren't we getting warnings  about not using the solver?!

File: tests/test_amgx_direct_solver.cu (hunk 1 above)
Comment (masterleinad) [reply tests/test_amgx_direct_solver.cu]: No, this wasn't necessarily wrong. The reasoning is just to not provide the solution as initial guess to the solver (unless we want to test for this corner case).
I first tested with `solution_dev` initialized by `rhs`, but then decided to go for zero-initialization to match the serial test case.
We could do
```
dealii::LinearAlgebra::distributed::Vector<double, dealii::MemorySpace::CUDA>
solution_dev(sol_ref.get_partitioner());
```
instead if you prefer that.

Comment (dalg24) [reply tests/test_amgx_direct_solver.cu]: Sure

File: tests/test_amgx_direct_solver.cu (hunk 2 above)
Comment (Rombur) [reply tests/test_amgx_direct_solver.cu]: No :(  I have seen that several times when it's a cuda file. I don't know if it's because of nvcc or if we are not passing the flags correctly.

**Review by Rombur — APPROVED** [review]

**Comment by Rombur** [comment]
This needs to be rebased.

**Comment by masterleinad** [comment]
Squashed and rebased.
